### PR TITLE
ISPDB update for bluewin.ch/bluemail.ch

### PR DESCRIPTION
**Pull request by goe-tautau**
- The two domains bluewin.ch and bluemail.ch now share a more aligned setup and can be handled via one set of configuration directives
- Bluewin and Bluemail are brands of Swisscom (Switzerland) Ltd, so the emailProvider has been changed to Swisscom
- Authentication method values have been synchronised between protocols
- The documentation URL has been updated (the new URL will redirect to whatever location provides the up-to-date documentation)
- The webmail config, as included before, is now defunct and has been removed. The current webmail credentials are unrelated to the mail addresses, so only the URL to the web mail entry page has been left

### CODE REVIEW

File: bluewin.ch [hunk 1]
```diff
@@ -1,41 +1,34 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <clientConfig version="1.1">
-  <emailProvider id="bluewin.ch">
+  <emailProvider id="Swisscom">
```
Comment (benbucksch) [inline bluewin.ch]: ID must be a domain

File: bluewin.ch [hunk 2]
```diff
@@ -1,41 +1,34 @@
...
       <authentication>password-cleartext</authentication>
     </incomingServer>
     <outgoingServer type="smtp">
       <hostname>smtpauths.bluewin.ch</hostname>
       <port>465</port>
       <socketType>SSL</socketType>
-      <username>%EMAILLOCALPART%</username>
-      <authentication>password-encrypted</authentication>
+      <username>%EMAILADDRESS%</username>
+      <authentication>password-cleartext</authentication>
```
Comment (benbucksch) [inline bluewin.ch]: I know this is SSL, but hashed passwords are always better than plaintext passwords.

Please try whether encrypted logins work. Try all of IMAP, POP3 and SMTP. If they work, then we want to use them, not plaintext. If they work only for SMTP, we keep it there.

File: bluewin.ch [hunk 3]
```diff
@@ -1,41 +1,34 @@
...
   <webMail>
-    <loginPage url="https://rich.sso.bluewin.ch/cp/applink/sso/Login?d=%EMAILDOMAIN%" />
-    <loginPageInfo
-      url="https://rich.sso.bluewin.ch/cp/applink/sso/Login?d=%EMAILDOMAIN%">
-      <username>%EMAILLOCALPART%</username>
-      <usernameField id="username" name="user" />
-      <passwordField name="password" />
-      <loginButton name="anmelden" />
-    </loginPageInfo>
+    <loginPage url="https://webmail.bluewin.ch/" />
```
Comment (benbucksch) [inline bluewin.ch]: Can the user no longer log in even with their email address and password?

Please fix the values rather than removing them.

Comment (benbucksch) [reply bluewin.ch]: FWIW, neither of these URLs, not even the documentation page, respond to me when I try to open them in a browser.

File: bluewin.ch (hunk 2 above)
Comment (goe-tautau) [reply bluewin.ch]: The encrypted password types that are currently still available are planned to be phased out this is in preparation of this step.
The auth mechanisms available don't allow optimal storage of credentials at rest and will thus be removed.

File: bluewin.ch (hunk 3 above)
Comment (goe-tautau) [reply bluewin.ch]: The URLs are tested and correct, they redirect to the correct pages and are the communicated entry points.
The login to webmail works via a different login which is used to manage your whole account, potentially including multiple mail addresses. This login is unrelated to the available mail addresses and can not be systematically correlated.

File: bluewin.ch (hunk 1 above)
Comment (goe-tautau) [reply bluewin.ch]: what is the preferred domain here then? The documentation [ConfigFileFormat](https://wiki.mozilla.org/Thunderbird:Autoconfiguration:ConfigFileFormat) doesn't specify anything specific.
The company domain that bluewin.ch & bluemail.ch belong to would be swisscom.ch though no customer mail is run over this domain.

File: bluewin.ch (hunk 2 above)
Comment (benbucksch) [reply bluewin.ch]: Are you working for Swisscom?

Please note that these configurations will only affect new users. They will not update old users. Thus, you cannot remove existing authentication mechanisms, unless you want to break all existing users.

File: bluewin.ch (hunk 3 above)
Comment (benbucksch) [reply bluewin.ch]: FWIW, the URLs now work for me. They did not 2 hours ago.

File: bluewin.ch (hunk 1 above)
Comment (benbucksch) [reply bluewin.ch]: swisscom.ch should be good, then. Per spec, customer domains are under the `<domain>` tags.

File: bluewin.ch (hunk 2 above)
Comment (goe-tautau) [reply bluewin.ch]: I am, and we are aware of that, the necessary steps to communicate these changes will be taken.

File: bluewin.ch (hunk 3 above)
Comment (goe-tautau) [reply bluewin.ch]: Interesting, these URLs haven't changed in a while, they just were never updated in the ISPDB

File: bluewin.ch (hunk 1 above)
Comment (goe-tautau) [reply bluewin.ch]: 3969ae7b475ab16eb52e55a2e819eb6551cddacb fixes this.
Is there more we should address?

File: bluewin.ch [hunk 4]
```diff
@@ -1,41 +1,34 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <clientConfig version="1.1">
-  <emailProvider id="bluewin.ch">
+  <emailProvider id="swisscom.ch">
     <domain>bluewin.ch</domain>
-    <!-- See also bluemail.ch -->
-    <displayName>bluewin.ch</displayName>
-    <displayShortName>bluewin.ch</displayShortName>
+    <domain>bluemail.ch</domain>
```
Comment (benbucksch) [inline bluewin.ch]: You need to add bluewin.ch as domain, if you want this config to apply to foo@bluewin.ch as well as foo@bluemail.ch :
```suggestion
    <domain>bluewin.ch</domain>
    <domain>bluemail.ch</domain>
```

File: bluewin.ch (hunk 3 above)
Comment (benbucksch) [reply bluewin.ch]: So, is there no way you can maintain the ability of user to log in with their email address and password to see their webmail?
If there is, could you please fix the form values?
If not, I would be interested to know why not.

File: bluewin.ch (hunk 4 above)
Comment (goe-tautau) [reply bluewin.ch]: <domain>bluewin.ch</domain> is on line 4 so just not visible in the PR view but it is already there

Comment (benbucksch) [reply bluewin.ch]: correct, sorry

File: bluewin.ch (hunk 3 above)
Comment (goe-tautau) [reply bluewin.ch]: Simply put: there is currently no way to login to Webmail with your email address and password, there hasn't been in years and it probably wont be possible in the foreseeable future.
It has all to do with Single Sign On, corporate decisions and account modelling but to go into the details here would probably help no one.

**Comment by benbucksch** [comment]
re password-encrypted: It pains me to see an encrypted authentication method be replaced with a plaintext one (even if via SSL), and it pains me even more to know that Swisscom will break all existing users for this.

But what  can I do?

**Comment by benbucksch** [comment]
All I can do is plea with you:

@goe-tautau: Please consider the pain that your users will have to go through when they find that they cannot send email anymore and need to manually reconfigure their accounts. I know you want to "communicate" that to them, but if you have ever watched a normal user (maybe 70 years old) try to configure an email account, you will stop and think that this is not a good idea to break them.

**Comment by goe-tautau** [comment]
@benbucksch I see your point, and rest assured this is not something we do lightly

**Comment by Sancus** [comment]
This is live now.

**Pull request merged**